optimistic_lock.hpp

```diff
@@ -199,15 +212,41 @@ class [[nodiscard]] optimistic_lock final {
       return *this;
     }
 
+    // Point to the same lock and have the same version for that lock.
+    bool operator==(const read_critical_section &other) const noexcept {
```

Comment:
```suggestion
    [[nodiscard]] bool operator==(const read_critical_section &other) const noexcept {
```
(and the header comment redundant?)

Reply:
I found this comment to be clarifying, leaving it.  Made the change to add [[nodiscard]].